ntoskrnl/io/pnpmgr/pnpres.c

```diff
@@ -959,13 +988,21 @@ IopTranslateDeviceResources(
    }
    RtlCopyMemory(DeviceNode->ResourceListTranslated, DeviceNode->ResourceList, ListSize);
 
+   FullDescriptor = &DeviceNode->ResourceList->List[0];
    for (i = 0; i < DeviceNode->ResourceList->Count; i++)
    {
-      pPartialResourceList = &DeviceNode->ResourceList->List[i].PartialResourceList;
+      pPartialResourceList = &FullDescriptor->PartialResourceList;
+      FullDescriptor = CmiGetNextResourceDescriptor(FullDescriptor);
+
       for (j = 0; j < pPartialResourceList->Count; j++)
       {
          DescriptorRaw = &pPartialResourceList->PartialDescriptors[j];
-         DescriptorTranslated = &DeviceNode->ResourceListTranslated->List[i].PartialResourceList.PartialDescriptors[j];
```

Comment:
Just a remember comment (for whoever is interested):
This old/broken construct is also used in our drivers:
https://git.reactos.org/?p=reactos.git;a=blob;f=drivers/parallel/parport/fdo.c;hb=99f979d6e9bd5c8f668e709754827fb99a8a3f80#l106
https://git.reactos.org/?p=reactos.git;a=blob;f=drivers/serial/serial/pnp.c;hb=99f979d6e9bd5c8f668e709754827fb99a8a3f80#l139
https://git.reactos.org/?p=reactos.git;a=blob;f=drivers/storage/floppy/fdc/fdo.c;hb=99f979d6e9bd5c8f668e709754827fb99a8a3f80#l86

Reply:
All of those are correct, since they check that ResourceList->Count is 1.